### PR TITLE
Streamlining test deps a bit

### PR DESCRIPTION
**Pull request by psfinaki**
This is just a tiny subset of what @majocha is doing [here](https://github.com/dotnet/fsharp/pull/17872) to help up a bit :)

### CODE REVIEW

File: tests/FSharp.Build.UnitTests/FSharp.Build.UnitTests.fsproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   <PropertyGroup>
     <TargetFrameworks>net472;$(FSharpNetCoreProductTargetFramework)</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' == 'Unix'">$(FSharpNetCoreProductTargetFramework)</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' == 'Unix' or '$(BUILDING_USING_DOTNET)' == 'true'">$(FSharpNetCoreProductTargetFramework)</TargetFrameworks>
```
Comment (psfinaki) [inline tests/FSharp.Build.UnitTests/FSharp.Build.UnitTests.fsproj:7]: That's for easier local development in `FSharp.sln`.

Comment (psfinaki) [reply tests/FSharp.Build.UnitTests/FSharp.Build.UnitTests.fsproj:7]: Yep, I meant it, typed the wrong thing 🤦

Comment (vzarytovskii) [reply tests/FSharp.Build.UnitTests/FSharp.Build.UnitTests.fsproj:7]: No, it's FCS solution:
https://github.com/dotnet/fsharp/blob/318e1c85a73428e6267781aef867f9f290b8bc18/Directory.Build.props#L24-L26

This flag isn't supported with any other solutions.

Comment (psfinaki) [reply tests/FSharp.Build.UnitTests/FSharp.Build.UnitTests.fsproj]: Alrighty, removed the change.

Comment (vzarytovskii) [reply tests/FSharp.Build.UnitTests/FSharp.Build.UnitTests.fsproj]: The change was fine, but the comment was not right :)

**Review by T-Gro — APPROVED** [review]

**Review by abonie — APPROVED** [review]

**Pull request merged**